src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -71,7 +71,8 @@ public boolean start() throws Exception {
         node.addAction(new PauseAction("Input"));
 
         String baseUrl = '/' + run.getUrl() + getPauseAction().getUrlName() + '/';
-        if (input.getParameters().isEmpty()) {
+        //JENKINS-40594 submitterParameter does not work without at least one actual parameter
+        if (input.getParameters().isEmpty() && (input.getSubmitterParameter() == null || input.getSubmitterParameter().isEmpty())) {
```

Comment:
Thanks for your help, I did tests again with your changes with below 3 test cases. It works well.
Yes, your guess is correct: this issue is caused by the console hyperlink in UI.
===========================================================================
def x = input message: 'Approve or not: case 1', submitter: 'admin,dev', submitterParameter: 'my-approval'
echo x
x = input message: 'Approve or not: case 2', submitterParameter: 'my-approval'
echo x
x = input message: 'Approve or not: case 3', submitter: '', submitterParameter: 'my-approval'
echo x
================ Here is output of console ========================
Started by user admin
[Pipeline] input
Input requested
Approved by admin
[Pipeline] echo
admin
[Pipeline] input
Input requested
Approved by admin
[Pipeline] echo
admin
[Pipeline] input
Input requested
Approved by admin
[Pipeline] echo
admin
[Pipeline] End of Pipeline
Finished: SUCCESS